plugins/database/memory/src/storage/node.ts

```diff
@@ -26,11 +26,11 @@ export class Storage {
     const files = await fs.readdir(root)
     await Promise.all(files.map(async (filename) => {
       const extension = extname(filename)
-      if (extension !== loader) return
-      const buffer = await fs.readFile(filename)
+      if (extension !== `.${loader}`) return
+      const buffer = await fs.readFile(resolve(root, filename))
       try {
         const data = await this.load(buffer, loader)
-        const name = filename.slice(0, -1 - extension.length)
+        const name = filename.slice(0, filename.length - extension.length)
```

Comment:
maybe 

```ts
filename.slice(0, -extension.length)
```

is better?

Reply:
I was considering that this will return an empty string if `extension` is empty but practically this never happens.

Reply:
If someone modified the code to allow using data files without an extension then the use of `-extension.length` will break things.

Reply:
That makes sense.